### PR TITLE
Update data exfiltration package

### PR DESCRIPTION
**Pull request by susan-shu-c**
## What does this PR do?

- [x] Remove the Technical Preview flag
- [x]  Change the title of package from DED to "Data Exfiltration Detection"
- [x]  Create and lens dashboard to visualize analytics - [specs](https://github.com/elastic/security-ml/issues/180#issuecomment-1334243361)
- [ ] Export the lens dashboard files into this package's source files
- [ ] Improve README re: rules

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

* https://github.com/elastic/security-ml/issues/180

### CODE REVIEW

File: packages/ded/kibana/ml_module/ded-ml.json [hunk 1]
```diff
@@ -238,7 +238,7 @@
                 "config": {
                     "groups": [
                         "security",
-                        "DED"
+                        "ded"
```
Comment (peteharverson) [inline packages/ded/kibana/ml_module/ded-ml.json:241]: Not introduced by this PR, but the `detector_description` of this job doesn't match the `over_field_name` - it should refer to `destination.geo.region_name`.

File: packages/ded/kibana/ml_module/ded-ml.json [hunk 2]
```diff
@@ -272,7 +272,7 @@
...
```
Comment (peteharverson) [inline packages/ded/kibana/ml_module/ded-ml.json:275]: Not introduced by this PR, but the `detector_description` of this job doesn't match the `over_field_name` - it should refer to `destination.geo.timezone`.

**Review by peteharverson — COMMENTED** [review]
The `groups` changes look good, but I noticed a couple of the detector descriptions are incorrect, so it would be good to update those in this PR too.

File: packages/ded/kibana/ml_module/ded-ml.json (hunk 1 above)
Comment (susan-shu-c) [reply packages/ded/kibana/ml_module/ded-ml.json:241]: Thank you for noting those! I'll make those changes 🙌

**Review by peteharverson — APPROVED** [review]
Changes to ML job groups and detector descriptions LGTM

**Comment by ajosh0504** [comment]
I removed more references to DED, and updated the package versions, manifest etc. Also tested these changes locally via `elastic-package` to make sure everything is working as expected.

1. Package updated to version `0.0.2`, and shows up as `Data Exfiltration Detection` instead of `DED`
Screen Shot 2022-12-12 at 9 06 01 PM

2. The ML job module shows up with the title `Data Exfiltration Detection` instead of DED
Screen Shot 2022-12-12 at 9 07 24 PM

3. All jobs were enabled successfully
Screen Shot 2022-12-12 at 9 08 48 PM

4. All rules were enabled, and are tagged `Data Exfiltration` instead of `DED`
Screen Shot 2022-12-12 at 9 05 25 PM

**Comment by susan-shu-c** [comment]
Note from sync: let's also update the old "optional" wording in the README, I thought I'd requested those changes in a previous PR but maybe that wasn't merged or was closed (would have been created by Omid)

File: packages/ded/docs/README.md [hunk 3]
```diff
@@ -1,26 +1,22 @@
...
+In **Machine Learning > Anomaly Detection**, when you create a job, you should see an option to `Use preconfigured jobs` with a card for Data Exfiltration Detection. When you select the card, you will see a pre-configured anomaly detection job that you can enable depending on what makes the most sense for your environment.
 
 ### (Optional) Enable Security rules
 
-In order to maximize the benefit of the DED detection framework, you might consider activating detection rules that are triggered when certain conditions for the anomaly detection jobs are satisfied. See the [documentation](https://www.elastic.co/guide/en/security/current/detection-engine-overview.html) for more information on importing and enabling the rules.
+To maximize the benefit of the Data Exfiltration Detection detection framework, you might consider activating detection rules that are triggered when certain conditions for the anomaly detection jobs are satisfied. See the [documentation](https://www.elastic.co/guide/en/security/current/detection-engine-overview.html) for more information on importing and enabling the rules.
```
Comment (szeitlin) [inline packages/ded/docs/README.md:17]: Don't forget to make this non-optional

**Review by szeitlin — APPROVED** [review]

**Comment by susan-shu-c** [comment]
## Update
(We won't merge until this is resolved)

_Exporting dashboards_

We started with the dashboard .json file, but encountered errors. Not all of these are important but documenting them for future references.
1. .DS_Store preventing `elastic-package` from building. Resolved by removing the file via command line.
2. Version mismatches. `elastic-package` (newest as of today) pulls 8.5.0 by default, but cloud is 8.6.0. Resolved by upgrading or editing the .json file.
3. Lens object not found. Though some packages (e.g. `akamai`) only have dashboards, some such as `azure` also have lens objects. Spoke to Taylor and he recommended we add the lens objects.
4. The dashboard file has a reference to index patterns. Looked at other packages but they don't have index pattern files objects. Asked Taylor but he's not sure either if it's supported (but that the package spec might not forbid it)? - Edit - not supported in package spec - screenshot 3
5. (Unresolved) Lens dashboard object not found. After adding lens files, having some sort of `state` error.

Screen Shot 2022-12-16 at 10 26 41 AM

_At least the the dashboards are "installed", but not working_
Screen Shot 2022-12-16 at 10 26 31 AM

_`index-pattern` not allowed in package_
Screen Shot 2022-12-16 at 11 09 06 AM

**Comment by ajosh0504** [comment]
We've decided to ship this package without dashboards for 8.6. We'll push an update around 8.7 once we have more clarity on what needs to happen to get the dashboards to work.

File: packages/ded/docs/README.md (hunk 3 above)
Comment (ajosh0504) [reply packages/ded/docs/README.md]: Done!

**Pull request merged**